### PR TITLE
Added some Google Cloud Platform explanations

### PR DESCRIPTION
**Pull request by simonw**
When I ran through this tutorial I wasn't clear on the reason for some of the commands, so I did some research and fleshed out this section with basic explanations and links to the relevant Google Cloud Platform documentation.

### CODE REVIEW

File: docs/01-infrastructure-gcp.md [hunk 1]
```diff
@@ -52,29 +57,50 @@ gcloud compute networks subnets create kubernetes \
...
+
+`10.240.0.0/24` means IPs from `10.240.0.0` to `10.240.0.254`.
+
 ### Create Firewall Rules
 
+https://cloud.google.com/compute/docs/vpc/firewalls
+
+A GCP network also acts as a firewall. By default no connections are allowed from the outside world, and connections between instances are also forbidden. We can add firewall rules to allow our instances to talk to each other within the network.
+
+This creates a rule called `allow-internal` which allows TCP, UDP and ICMP connections between all machines in your `10.240.0.0/24` subnet, and also enables those machines to talk to the CIDR range `10.200.0.0/16`:
```
Comment (simonw) [inline docs/01-infrastructure-gcp.md]: I have to admit I don't understand the need for the 10.200.0.0/16 range - I managed to figure out it was something to do with the acronym CIDR but I don't understand why it's needed or what effect it has.

Comment (harshal-shah) [reply docs/01-infrastructure-gcp.md]: @simonw The 10.200.0.0/16 range is the set of IP's from which your pods will get their IP addresses. 
This range is passed as a parameter to kubernetes-controller-manager https://github.com/kelseyhightower/kubernetes-the-hard-way/blob/master/docs/05-kubernetes-controller.md

More details on controller manager flags : https://kubernetes.io/docs/admin/kube-controller-manager/

Comment (simonw) [reply docs/01-infrastructure-gcp.md]: Fantastic! Thanks @harshal-shah that's exactly what I needed to know.

**Comment by kelseyhightower** [comment]
I've made a big overhaul to these docs with more details. I'm not sure if I've covered everything for you well enough, but please take another look.

**Pull request closed without merging**